packages/gateway-client/src/worker/p2p-fetch/p2p-fetch-request.ts

```diff
@@ -257,7 +257,8 @@ export class P2pFetchRequest {
         const p2pRequest = new P2pRequest(
             this.requestId,
             this.p2pClient,
-            packet
+            packet,
+            this.reqInit.signal ?? undefined
```

Comment:
@rynomad I wanted our fetch method to support abort signals (like the native fetch does). We'll need to re-implement this in your PR that refactors the p2p fetch logic (hopefully a bit more cleanly than I did here).